resources/environments/schema_azure_environment.go

```diff
@@ -184,6 +184,42 @@ var AzureEnvironmentSchema = schema.Schema{
 				},
 			},
 		},
+		"compute_cluster": schema.SingleNestedAttribute{
+			MarkdownDescription: "Option to set up Externalized compute cluster for the environment.",
+			Optional:            true,
+			Attributes: map[string]schema.Attribute{
+				"enabled": schema.BoolAttribute{
+					Required: true,
+				},
+				"configuration": schema.SingleNestedAttribute{
+					MarkdownDescription: "The Externalized k8s configuration for the environment.",
+					Optional:            true,
+					Attributes: map[string]schema.Attribute{
+						"private_cluster": schema.BoolAttribute{
+							MarkdownDescription: "If true, creates private cluster. False, if not specified",
+							Default:             booldefault.StaticBool(false),
+							Computed:            true,
+							Optional:            true,
+						},
+						"kube_api_authorized_ip_ranges": schema.SetAttribute{
+							MarkdownDescription: "Kubernetes API authorized IP ranges in CIDR notation. Mutually exclusive with privateCluster.",
+							ElementType:         types.StringType,
+							Optional:            true,
+						},
+						"outbound_type": schema.StringAttribute{
```

Comment:
If it has a single possible value, why not default? Or preferably, just hide its existence and set it automatically.